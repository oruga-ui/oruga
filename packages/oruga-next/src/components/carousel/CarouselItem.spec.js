import { mount } from '@vue/test-utils'
import OCarousel from '@components/carousel/Carousel'
import OCarouselItem from '@components/carousel/CarouselItem'

let wrapper
let wrapperParent

const WrapperComp = {
    data() {
        return {}
    },
    template: `
        <OCarousel ref="carousel">
            <OCarouselItem value="item1"/>
            <OCarouselItem ref="testItem" value="item2"/>
            <OCarouselItem value="item3"/>
        </OCarousel>`,
    components: {
        OCarousel, OCarouselItem
    }
}

describe('OCarouselItem', () => {
    beforeEach(() => {
        wrapperParent = mount(WrapperComp, { sync: false })
        wrapper = wrapperParent.find({ ref: 'testItem' })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('OCarouselItem')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('computes its position correctly', () => {
        expect(wrapper.vm.index).toBe(1)
    })
})
