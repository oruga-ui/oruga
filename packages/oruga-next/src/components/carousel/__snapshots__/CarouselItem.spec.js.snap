// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OCarouselItem render correctly 1`] = `<div class="o-car__item" style="width: 0px;" value="item2"></div>`;
